rust/flatbuffers/src/vector.rs

```diff
@@ -135,3 +146,45 @@ impl<'a, T: Follow<'a> + 'a> Follow<'a> for Vector<'a, T> {
     }
 }
 
+pub struct VectorIter<'a, T:'a >(Vector<'a, T>, usize);
+impl<'a, T: 'a> VectorIter<'a, T> {
+    pub fn new(inner: Vector<'a, T>) -> Self {
+        VectorIter(inner, 0)
+    }
+}
+
+impl<'a, T: Follow<'a> + 'a> Clone for VectorIter<'a, T> {
+    fn clone(&self) -> Self {
+        VectorIter(self.0, self.1)
+    }
+}
+
+impl<'a, T: Follow<'a> + 'a> Iterator for VectorIter<'a, T> {
```

Comment:
@jaynus We should provide a [`size_hint`] using vector's [`len`] method.

[`size_hint`]: https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.size_hint
[`len`]: https://docs.rs/flatbuffers/0.6.0/flatbuffers/struct.Vector.html#method.len